src/shogun/lib/Signal.cpp

```diff
@@ -32,9 +32,9 @@ CSignal::~CSignal()
 
 void CSignal::handler(int signal)
 {
-	/* If the handler is not enabled, then return */
+	/* If the handler is not enabled exit */
 	if (!m_active)
-		return;
+		exit(-1);
```

Comment:
are we sure we want exit(-1) ? i mean in this case of a REPL we would exit the REPL itself, or?

Reply:
The user won't enter into REPL loop if m_active is set to false. In such a situation, the handler won't ask the user any inputs, it will just terminate the execution and it won't print any message to the screen either. So I decided to maintain a consisten behaviour between the Shogun's handler and the standard system handler when dealing with a SIGINT: exit and set the exit code != 0.